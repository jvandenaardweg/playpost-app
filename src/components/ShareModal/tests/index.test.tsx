import React from 'react';
import { render, RenderAPI } from 'react-native-testing-library';

import { ShareModal } from '../index';

describe('ShareModal', () => {
  let wrapper: RenderAPI;
  const onPressCloseHandler = jest.fn();
  const errorMessage = 'Test error message';
  const successMessage = 'Article is added to your playlist!';

  describe('rendering', () => {
    beforeAll(() => {
      wrapper = render(<ShareModal isLoading={false} isSuccess={false} isError={false} errorMessage={''} onPressClose={onPressCloseHandler} />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should show an error message when errorMessage and isError is set', () => {
      wrapper.update(<ShareModal isLoading={false} isSuccess={false} isError={true} errorMessage={errorMessage} onPressClose={onPressCloseHandler} />);
      expect(wrapper.queryByText(errorMessage)).toBeTruthy();
    });

    it('should not show an error message when isLoading is set', () => {
      wrapper.update(<ShareModal isLoading={true} isSuccess={false} isError={true} errorMessage={errorMessage} onPressClose={onPressCloseHandler} />);
      expect(wrapper.queryByText(errorMessage)).toBeFalsy();
    });

    it('should not show an success message when isLoading is set', () => {
      wrapper.update(<ShareModal isLoading={true} isSuccess={false} isError={false} errorMessage={errorMessage} onPressClose={onPressCloseHandler} />);
      expect(wrapper.queryByText(successMessage)).toBeFalsy();
    });

    it('should show an success message when isSuccess is set', () => {
      wrapper.update(<ShareModal isLoading={false} isSuccess={true} isError={false} errorMessage={''} onPressClose={onPressCloseHandler} />);
      expect(wrapper.queryByText(successMessage)).toBeTruthy();
    });

  });

  describe('interacting', () => {
    beforeAll(() => {
      wrapper = render(<ShareModal isLoading={false} isSuccess={false} isError={false} errorMessage={''} onPressClose={onPressCloseHandler} />);
    });

    it('should fire onPressClose when pressing the close button', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
