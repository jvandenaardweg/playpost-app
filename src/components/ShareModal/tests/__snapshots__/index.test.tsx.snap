// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareModal interacting should fire onPressClose when pressing the close button 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 10,
      "elevation": 10,
      "height": 168,
      "padding": 12,
      "shadowColor": "#040404",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 3.84,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 6,
          "justifyContent": "center",
          "minHeight": 70,
          "textAlign": "center",
        }
      }
    />
    <View
      style={Object {}}
    >
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="ShareModal-Button-close"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#2089dc",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Close
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`ShareModal rendering should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 10,
      "elevation": 10,
      "height": 168,
      "padding": 12,
      "shadowColor": "#040404",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 3.84,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 6,
          "justifyContent": "center",
          "minHeight": 70,
          "textAlign": "center",
        }
      }
    />
    <View
      style={Object {}}
    >
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="ShareModal-Button-close"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#2089dc",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Close
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
